src/shared/components/oncoprint/controls/OncoprintControls.tsx

```diff
@@ -207,9 +221,33 @@ export default class OncoprintControls extends React.Component<
 > {
     @observable horzZoomSliderState: number;
     @observable heatmapGenesReady = false;
-    @observable private _selectedTreatmentIds: string[] = [];
-    private textareaTreatmentText = '';
-    @observable treatmentFilter = '';
+    @observable private _selectedGenericAssayEntityIds: string[] =
+        this.props.selectedGenericAssayEntitiesGroupByGenericAssayTypeFromUrl &&
```

Comment:
this belongs in constructor

Reply:
and in some kind of routine named initialization

Reply:
why not @computed?